web/src/localstorage.ts

```diff
@@ -109,15 +106,17 @@ const ls = {
         keys = keys.filter((key) => key_regex.test(key));
 
         for (const key of keys) {
-            let value;
-            let value_set = false;
+            let raw_data;
             try {
-                value = localStorage.getItem(key);
-                value_set = true;
+                raw_data = localStorage.getItem(key);
             } catch {
-                // Do nothing if the fetch fails
+                continue;
             }
-            if (value_set && condition_checker(value)) {
+            if (raw_data === null) {
+                continue;
+            }
+            const data = formDataSchema.parse(JSON.parse(raw_data));
+            if (ls.isExpired(data.expires) || condition_checker(data.data)) {
```

Comment:
Is removing expired items the right thing to do? It’s odd that the rest of the module makes no effort to do that at all; it just leaves expired items in place and refuses to read them.

Do we even use the expiration feature? This whole system is overengineered.

Reply:
I'd be happy to see us remove the `expires` feature if it's indeed unused. I think the only useful part of 8b22b94ab1613f523be01572787a13e3d49c50fd was doing JSON marshalling, which is pretty minimal value.

Reply:
I think removing them is likely correct; the only reason to not remove bad or expired data would be if we're worried we might have a temporary bug in the code for making that determination.

Reply:
- #31648